src/averageAges.js

```diff
@@ -37,7 +42,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womanWithChild = withChildren
+    ? people.filter(allPeople => people
+      .some(woman => woman.mother === allPeople.name))
+    : people.filter(person => person.sex === 'f');
+  const agelife = womanWithChild.map(x => x.died - x.born);
```

Comment:
```suggestion
  const ageLife = womanWithChild.map(x => x.died - x.born);
```
do not forget about CamelCase